### PR TITLE
Fix #18364: No Too High! or Too Low! error messages when attempting to raise/lower land beyond the limits

### LINKED ISSUE #18364: No "Too high!" or "Too low!" error messages when attempting to raise/lower land beyond the limits
### Operating System

Windows 10, 64-bit

### OpenRCT2 build

OpenRCT2, v0.4.2 (8ceea45) provided by GitHub

### Describe the issue

Oceanic Happyland 2022-10-16 22-35-49
When you attempt to raise or lower the land or water level beyond the limit, it normally blocks you with an error message saying either "Too high!" or "Too low!". However, neither of these error messages appear. In the above screenshot, I tried to lower the land below the lower limit of -30 ft (-9 m, -6 units), and it fails, but it doesn't show "Too low!" after the "Can't lower land here..." in the error message.
On a side note, the "Too low!" and "Too high!" error messages still appear in ride construction.

### Area(s) with issue?

_No response_

### Steps to reproduce

1. Raise or lower land or water level to the highest or lowest possible limit, respectively
2. Attempt to raise or lower it any further
3. No "Too high!" or "Too low!" error message will appear

### Attachments

_No response_

### PR DESCRIPTION
**Pull request by HouseholdVTuber**
Added missing ErrorMessage line in respective height range-related Error results for LandRaiseAction and LandLowerAction

This is my first contribution so I also appended my user to the Bug Fixes list. Thank you!

### CODE REVIEW

**Review by Gymnasiast — DISMISSED** [review]
This overwrites several other errors, e.g. “Forbidden by local authority”.

I did some digging and I saw where the problem lies. Do you want me to tell you, or would you prefer to find it out yourself?

**Comment by HouseholdVTuber** [comment]
If you've already found it then I'd appreciate your answer, thanks! I'd seen the existing check for whether it was outside of the property and figured all else was accounted for.

**Comment by HouseholdVTuber** [comment]
In the meantime I amended my change to only return that ErrorMessage if the new height was at or above/below the valid range, but it looks like the ride height code handles these GameActions::Status errors differently. I'll spend some more time with it until I hear otherwise. Thanks again.

**Comment by Broxzier** [comment]
This is not the place to add that check. Have a look at where the result object comes from. `GameActions::ExecuteNested` and `GameActions::QueryNested`.

**Comment by HouseholdVTuber** [comment]
Okay I'm starting to see it, so the result's ErrorTitle is already being returned as, "Too Low!" from the logic in LandSetHeightAction which already checked against the min/max height. But when we're lowering or raising land we want that to be in the result's ErrorMessage value, after the ErrorTitle which would expectedly say, "Can't Lower/Raise Land Here..."? So the information's just being overwritten.

**Comment by HouseholdVTuber** [comment]
Alright I'm much happier with this, the other attempts feel comparatively shortsighted. Thank you for your advice Gymnasiast and Broxzier. 

I'm trying to adhere to the code formatting style but it looks like I'm missing something, I'm going to run clang-format locally and figure it out before attempting more commits.

**Comment by HouseholdVTuber** [comment]
I can also expand on this to account for the other errors like Gymnasiast mentioned if this implementation is acceptable, either now or in a separate issue depending on how you guys like to track things.

File: src/openrct2/actions/LandSetHeightAction.cpp [hunk 1]
```diff
@@ -55,6 +55,14 @@ GameActions::Result LandSetHeightAction::Query() const
...
     if (errorTitle != STR_NONE)
     {
+        if (errorTitle == STR_TOO_LOW)
+        {
+            return GameActions::Result(GameActions::Status::TooLow, errorTitle, STR_NONE);
+        }
+        if (errorTitle == STR_TOO_HIGH)
+        {
+            return GameActions::Result(GameActions::Status::TooHigh, errorTitle, STR_NONE);
+        }
```
Comment (Broxzier) [inline src/openrct2/actions/LandSetHeightAction.cpp:65]: Have a look at what `errorTitle` holds, and how `LandSetHeightAction::CheckParameters` works.

Not directly related to this PR: both of these identifiers could be renamed to something that makes more sense - `ValidateParameters` for instance. 'Check' is a useless term in programming.

File: src/openrct2/actions/LandLowerAction.cpp [hunk 2]
```diff
@@ -122,6 +122,11 @@ GameActions::Result LandLowerAction::QueryExecute(bool isExecuting) const
             else
             {
                 result.ErrorTitle = STR_CANT_LOWER_LAND_HERE;
+                if (result.Error == GameActions::Status::TooLow)
+                {
+                    result.ErrorMessage = STR_TOO_LOW;
+                }
+
```
Comment (Broxzier) [inline src/openrct2/actions/LandLowerAction.cpp:129]: The error message should not be set here.

File: src/openrct2/actions/LandRaiseAction.cpp [hunk 3]
```diff
@@ -118,6 +118,11 @@ GameActions::Result LandRaiseAction::QueryExecute(bool isExecuting) const
             else
             {
                 result.ErrorTitle = STR_CANT_RAISE_LAND_HERE;
+                if (result.Error == GameActions::Status::TooHigh)
+                {
+                    result.ErrorMessage = STR_TOO_HIGH;
+                }
+
```
Comment (Broxzier) [inline src/openrct2/actions/LandRaiseAction.cpp:125]: Same here

**Review by Broxzier — CHANGES_REQUESTED** [review]

**Comment by HouseholdVTuber** [comment]
Hah, so I was apprehensive about changing what LandSetHeightAction::CheckParameters returns - but the only actions referencing it seem to be LandLowerAction, LandRaiseAction, and LandSmoothAction. Does this make sense? And... should I make it so that LandSetHeightAction also sets the return error title, and get rid of the lines in LandLowerAction/LandSetHeightAction that set the title? I'm still worried about changing too much.

Thanks again and I appreciate your patience, I made more complicated than it had to be.

File: src/openrct2/actions/LandSetHeightAction.cpp [hunk 4]
```diff
@@ -86,7 +86,7 @@ GameActions::Result LandSetHeightAction::Query() const
     // Check for ride support limits
     if (!gCheatsDisableSupportLimits)
     {
-        errorTitle = CheckRideSupports();
+        StringId errorTitle = CheckRideSupports();
```
Comment (Gymnasiast) [inline src/openrct2/actions/LandSetHeightAction.cpp:89]: This also needs to be the error message, rather than the title.

**Comment by HouseholdVTuber** [comment]
Latest commit returns a Result with the appropriate errorMessage instead of errorTitle in LandSetHeightAction.cpp, per Gymnasiast's review

**Review by Broxzier — APPROVED** [review]
Tested and works as expected.

I'd like to note that the way the parameters are validated and returned as error messages is not very consistent, but cleaning that up is something for another PR.

**Comment by HouseholdVTuber** [comment]
Thanks Broxzier, I can submit that as an issue and work on it later if that makes sense.

And would it be better for me to re-commit the contributors.md with its latest or just let someone with write access resolve the merge conflict?

**Comment by tupaschoal** [comment]
> And would it be better for me to re-commit the contributors.md with its latest or just let someone with write access resolve the merge conflict?

If you can solve the merge conflict, that's better. If you don't know how to do it, let us know and we can help

**Comment by HouseholdVTuber** [comment]
Thanks tupaschoal, I figured I just had to copy the latest and re-append my line for it to go through since it's just a text file

**Comment by Broxzier** [comment]
Great! That's usually the best approach with simple merge conflicts.

**Comment by tupaschoal** [comment]
Unfortunately there is still a merge conflict on "contributors.md", can you try again?

This tutorial might help you: https://github.com/OpenRCT2/OpenRCT2/wiki/Rebase-and-Sync-fork-with-OpenRCT2

**Comment by HouseholdVTuber** [comment]
Thanks for the link, I've got it now!!

**Review by tupaschoal — APPROVED** [review]

**Comment by tupaschoal** [comment]
@Gymnasiast can you review this one so we can merge it, please?

**Pull request merged**